api/envoy/config/core/v3/protocol.proto

```diff
@@ -157,6 +157,16 @@ message Http1ProtocolOptions {
   //   - Not a response to a HEAD request.
   //   - The content length header is not present.
   bool enable_trailers = 5;
+
+  // Allows Envoy to process requests with `Content-Length` and `Transfer-Encoding: chunked`.
+  // By default such requests are rejected with `400 Bad Request`, but if option is enabled -
```

Comment:
I'm not sure if I agree with the default here. Yes, if both are present, then `Transfer-Encoding` takes a precedence, and `Content-Length` may be removed, but rejecting such request seems a bit harsh.

Reply:
That's the current behavior (http-parser fails parsing with HPE_UNEXPECTED_CONTENT_LENGTH which ends up as Bad Request to the downstream), I'm keeping it consistent.